Mathlib/Analysis/Distribution/ContDiffMapSupportedIn.lean

```diff
@@ -0,0 +1,263 @@
+    extends FunLike B E F where
+  map_contDiff (f : B) : ContDiff ℝ n f
+  map_zero_on_compl (f : B) : EqOn f 0 Kᶜ
+
+open ContDiffMapSupportedInClass
+
+instance (B : Type*) (E F : outParam <| Type*)
+    [NormedAddCommGroup E] [NormedAddCommGroup F] [NormedSpace ℝ E] [NormedSpace ℝ F]
+    (n : outParam ℕ∞) (K : outParam <| Compacts E)
+    [ContDiffMapSupportedInClass B E F n K] :
+    ContinuousMapClass B E F where
+  map_continuous f := (map_contDiff f).continuous
+
+instance (B : Type*) (E F : outParam <| Type*)
+    [NormedAddCommGroup E] [NormedAddCommGroup F] [NormedSpace ℝ E] [NormedSpace ℝ F]
+    (n : outParam ℕ∞) (K : outParam <| Compacts E)
+    [ContDiffMapSupportedInClass B E F n K] :
+    BoundedContinuousMapClass B E F where
+  map_bounded f := by
+    have := HasCompactSupport.intro K.isCompact (map_zero_on_compl f)
+    rcases (map_continuous f).bounded_above_of_compact_support this with ⟨C, hC⟩
+    exact map_bounded (BoundedContinuousFunction.ofNormedAddCommGroup f (map_continuous f) C hC)
+
+namespace ContDiffMapSupportedIn
+
+instance toContDiffMapSupportedInClass :
+    ContDiffMapSupportedInClass 𝓓^{n}_{K}(E, F) E F n K where
+  coe f := f.toFun
+  coe_injective' f g h := by cases f; cases g; congr
+  map_contDiff f := f.contDiff'
+  map_zero_on_compl f := f.zero_on_compl'
+
+variable {E F}
+
+protected theorem contDiff (f : 𝓓^{n}_{K}(E, F)) : ContDiff ℝ n f := map_contDiff f
+protected theorem zero_on_compl (f : 𝓓^{n}_{K}(E, F)) : EqOn f 0 Kᶜ := map_zero_on_compl f
+protected theorem compact_supp (f : 𝓓^{n}_{K}(E, F)) : HasCompactSupport f :=
+  .intro K.isCompact (map_zero_on_compl f)
+
+@[simp]
+theorem toFun_eq_coe {f : 𝓓^{n}_{K}(E, F)} : f.toFun = (f : E → F) :=
+  rfl
+
+/-- See note [custom simps projection]. -/
+def Simps.apply (f : 𝓓^{n}_{K}(E, F)) : E →F  := f
+
+-- this must come after the coe_to_fun definition.
+initialize_simps_projections ContDiffMapSupportedIn (toFun → apply)
+
+@[ext]
+theorem ext {f g : 𝓓^{n}_{K}(E, F)} (h : ∀ a, f a = g a) : f = g :=
+  DFunLike.ext _ _ h
+
+/-- Copy of a `BoundedContDiffMap` with a new `toFun` equal to the old one. Useful to fix
+definitional equalities. -/
+protected def copy (f : 𝓓^{n}_{K}(E, F)) (f' : E → F) (h : f' = f) : 𝓓^{n}_{K}(E, F) where
+  toFun := f'
+  contDiff' := h.symm ▸ f.contDiff
+  zero_on_compl' := h.symm ▸ f.zero_on_compl
+
+@[simp]
+theorem coe_copy (f : 𝓓^{n}_{K}(E, F)) (f' : E → F) (h : f' = f) : ⇑(f.copy f' h) = f' :=
+  rfl
+
+theorem copy_eq (f : 𝓓^{n}_{K}(E, F)) (f' : E → F) (h : f' = f) : f.copy f' h = f :=
+  DFunLike.ext' h
+
+@[simp]
+theorem toBoundedContinuousFunction_apply (f : 𝓓^{n}_{K}(E, F)) (x : E) :
+   (f : BoundedContinuousFunction E F) x  = (f x) := rfl
+
+theorem _root_.Set.EqOn.comp_left₂ {α β δ γ} {op : α → β → δ} {a₁ a₂ : γ → α}
+    {b₁ b₂ : γ → β} {s : Set γ} (ha : s.EqOn a₁ a₂) (hb : s.EqOn b₁ b₂) :
+    s.EqOn (fun x ↦ op (a₁ x) (b₁ x)) (fun x ↦ op (a₂ x) (b₂ x)) := fun _ hx =>
+  congr_arg₂ _ (ha hx) (hb hx)
```

Comment:
Could you move this right after [Set.EqOn.comp_left](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Data/Set/Function.html#Set.EqOn.comp_left)?